lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -459,37 +452,48 @@ def access_token_url
       end
 
       def setup_access_token
-        request = 'grant_type=client_credentials'
-        begin
-          raw_response = ssl_post(access_token_url, request, access_token_header)
-        rescue ResponseError => e
-          raise OAuthResponseError.new(e)
-        else
-          response = parse(raw_response)
+        response = parse(ssl_post(access_token_url, 'grant_type=client_credentials', access_token_header))
+        @options[:access_token] = response['access_token']
+        @options[:expires] = response['expires_in'] && response['expires_in'] > 0 && (DateTime.now + (response['expires_in'] - 120).seconds).strftime('%Q').to_i
 
-          if (access_token = response['access_token'])
-            access_token
-          else
-            raise OAuthResponseError.new(response)
-          end
-        end
+        Response.new(
+          response['access_token'].present?,
+          message_from(true, response, {}),
```

Comment:
Just another small issue is that this should only be true if  response['access_token'].present?

Reply:
Will I use `access_token_valid?` as a verifier since we assign parts of the `response` that allow us to validate the presence of the `access_token` and the `expires`

Reply:
Ok, you just also have to update `message_from(true, response, {})` to be `message_from(access_token_valid?, response, {})`

Reply:
done!